### PR TITLE
[CST-4882] The search loads indefinitely if the written query is invalid

### LINKED ISSUE #1365: The search loads indefinitely if the written query is invalid
**Describe the bug**
The search that can be done from the search page and from the mydspace does not produce results, and loads indefinitely, if the query entered in the input is not valid. 
Example: https://demo7.dspace.org/search?spc.page=1&query=test:&spc.sf=score&spc.sd=DESC

**To Reproduce**
1. Search for a string like 'test:' or for any other string that does not represent a syntactically valid query

**Expected behavior**
At least one error message should be shown when this problem occurs.
It should also be considered whether to allow the user to search by escaping the characters entered in the input or using the DisMax Query Parser (https://solr.apache.org/guide/8_1/the-dismax-query-parser.html). A search of this type could be useful for example if you want to search for a word ending with ':' .

### PR DESCRIPTION
**Pull request by davidenegretti-4science**
## References
* Fixes #1365 
* Requires DSpace/DSpace#8044

## Description
Show error message if the search query fails. Handle error 422 (invalid query) separately.

## Instructions for Reviewers

If search fails then an error message should be shown instead of the loading component-

If search query is invalid (e.g. `test:`) the request fails and returns an error code 422. In this case the following message error is shown:  

> Search query is not valid. Please check [Solr query syntax](https://solr.apache.org/guide/query-syntax-and-parsing.html) best practices for further information about this error.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Comment by mspalti** [comment]
I removed search filters and no longer see the search-filter error in console.  But unfortunately my search ('test:') is still hanging and I don't see the error message.

**Review by mspalti — COMMENTED** [review]
This works as expected for My DSpace searches. It looks like the same fix could be applied to the main search component as well.  I don't know if that's out of scope for this PR or just an oversight. If out of scope it should be easy to fix in a separate PR. It needs to happen.

File: src/app/my-dspace-page/my-dspace-results/my-dspace-results.component.html [hunk 1]
```diff
@@ -10,5 +10,5 @@
   </ds-viewable-collection>
 </div>
 <ds-loading *ngIf="isLoading()" message="{{'loading.mydspace-results' | translate}}"></ds-loading>
-<ds-error *ngIf="searchResults?.hasFailed && (!searchResults?.errorMessage || searchResults?.statusCode != 400)" message="{{'error.search-results' | translate}}"></ds-error>
+<ds-error *ngIf="showError(searchResults)" message="{{errorMessageLabel(searchResults) | translate}}"></ds-error>
```
[the remark this answers is not in the document]
Comment (mspalti) [reply src/app/my-dspace-page/my-dspace-results/my-dspace-results.component.html:13]: One other brief comment. Could we get rid of the search-filter.component console error in this case by checking whether "payload" is `undefined` as well as the checking the totalElements count?  e.g.

`return valuesRD.payload && valuesRD.payload.totalElements > 0;`

[the remark this answers is not in the document]
Comment (davidenegretti-4science) [reply src/app/my-dspace-page/my-dspace-results/my-dspace-results.component.html]: @mspalti I fixed the console error and I also committed some additional tests that for some reason didn't work when I opened this PR. I'll fix the search component ASAP

**Review by tdonohue — CHANGES_REQUESTED** [review]
@davidenegretti-4science : I have the same results as @mspalti .  This fixes the bug for MyDSpace searches (`/mydspace`), but the original bug for the Search (`/search`) is still reproducible.  So, #1365 is not fully fixed.

Once the search component has been fixed, then we can retest this. Thanks!

**Comment by davidenegretti-4science** [comment]
@mspalti @tdonohue  the search component has been fixed

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me now too. Thanks @davidenegretti-4science !  Tested with https://github.com/DSpace/DSpace/pull/8044 and it works

**Pull request merged**